### DISCUSSION TITLE
Intermittent cursor jitters in-game while using opentabletdriver standalone or built-in

### DISCUSSION THREAD

**Discussion by Fleimi** [ppy/osu#19728] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Platform: Linux

Occasionally, the cursor will quickly teleport/jitter a bit when using either the standalone opentabletdriver, or the built-in one. I believe this to be a lazer bug, as the standalone driver works flawlessly on osu!stable, as well as on desktop. Mouse does not display this behaviour. Additionally, the wacom driver included with the kernel works with no issues.

### Screenshots or videos

https://user-images.githubusercontent.com/73422514/184423901-b131811b-d55e-4e35-a831-2aacb8c8bbe0.mp4

![osu_2022-08-12_21-46-09](https://user-images.githubusercontent.com/73422514/184423943-13d44290-25b8-430b-b783-1cc11b69e002.jpg)

### Version

2022.810.2-lazer

### Logs

[runtime.log](https://github.com/ppy/osu/files/9329236/runtime.log)
[network.log](https://github.com/ppy/osu/files/9329238/network.log)
[performance.log](https://github.com/ppy/osu/files/9329239/performance.log)
[database.log](https://github.com/ppy/osu/files/9329240/database.log)

**Comment by peppy** [comment, marked as answer]
The only thing that I can imagine causing this is having your tablet reporting very a two input methods. Try turning off high precision mouse, touch and tablet support in game and see if it still persists.

**Reply by Fleimi** [reply]
Hey,
Those options are already off. Of course when using the built-in tablet support/driver, I am disabling the standalone opentabletdriver.

**Reply by peppy** [reply]
Can you confirm that when disabling the external OTD, your tablet does NOT move your windows cursor?

**Reply by Fleimi** [reply]
I'm on linux as indicated by the post, but yes, I am unable to move my cursor in the desktop environment.

**Reply by peppy** [reply]
Oh my god. If it's linux then I have no idea, sorry. It could be absolutely anything or everything.

I thought you were on windows.

You'll need to figure if the issue is:
- Your system
- SDL
- Your WM
- Anything else in between

Then report it to the relevant party.

**Reply by Fleimi** [reply]
Thanks, I will do some digging and see whether I can suss out the culprit if you don't believe it's a lazer bug.

**Reply by TheFantasticLoki** [reply]
I'd like to add something. I do not believe this to be an issue with lazer itself and moreso OTD as I have been experiencing this issue intermittently over the past 2 years on stable. And it may also not be otd itself and actually the users tablet especially if they are using a small area and not full area. My tablet usually starts to show this issue more often if I have not moved my area on my tablet for a long time as I use no cover so the wear on the tablet itself has started to cause these random jitters at least in my own experience. 

EDIT: It's also very interesting that this issue does not appear on your tablets OEM driver, do you know if that driver is adding ANY form of smoothing to the input from the tablet as this would mask that issue? If it's genuinely a problem for you, you can add a smoothing filter to your tablet inside the OTD driver at the cost of some input delay but this should prevent that jitter from occurring with the correct settings

**Reply by Fleimi** [reply]
Yup, turns out it was my pen causing the issue. I disassembled it in its entirety to clean some gunk out from behind the button and after reassembling the problem went away completely

I suspect OTDs total lack of smoothing combined with lazer's extremely optimized and quick input handling resulted in the artifacts becoming apparent, whereas osu!stable, with its less robust input handling managed to mask it. Similarly, the driver included with the kernel has a ton of smoothing, masking the issue in both stable _and_ lazer.

**Comment by TheLowlyBard** [comment]
I'm a couple years late but just ran into a similar problem with my tablet. I had my tablet on top of my Logitech Powerplay mouse pad (charges your mouse wirelessly) and realized the RF emitted from the mousepad was messing with my pen's connection to the tablet.

**Reply by Rekunan** [reply]
That sounds like it should be a separate issue post in the [OTD GitHub](https://github.com/OpenTabletDriver/OpenTabletDriver).

**Reply by dragged9698** [reply]
oh my lord, godsend you have saved me so much trouble, just got the tablet in and was losing my mind. THANK YOU KIND SIR!!
